x-pack/plugins/case/server/client/cases/update.ts

```diff
@@ -10,36 +10,331 @@ import { pipe } from 'fp-ts/lib/pipeable';
+  comment: SavedObject<CommentAttributes>,
+  type: typeof CASE_SAVED_OBJECT | typeof SUB_CASE_SAVED_OBJECT
+): string | undefined {
+  return comment.references.find((ref) => ref.type === type)?.id;
+}
+
+/**
+ * Gets all the alert comments (generated or user alerts) for the requested cases.
+ */
+async function getAlertComments({
+  casesToSync,
+  caseService,
+  client,
+}: {
+  casesToSync: ESCasePatchRequest[];
+  caseService: CaseServiceSetup;
+  client: SavedObjectsClientContract;
+}): Promise<SavedObjectsFindResponse<CommentAttributes>> {
+  const idsOfCasesToSync = casesToSync.map((casePatchReq) => casePatchReq.id);
+
+  // getAllCaseComments will by default get all the comments, unless page or perPage fields are set
+  return caseService.getAllCaseComments({
+    client,
+    id: idsOfCasesToSync,
+    includeSubCaseComments: true,
+    options: {
+      filter: `${CASE_COMMENT_SAVED_OBJECT}.attributes.type: ${CommentType.alert} OR ${CASE_COMMENT_SAVED_OBJECT}.attributes.type: ${CommentType.generatedAlert}`,
+    },
+  });
+}
+
+/**
+ * Returns a map of sub case IDs to their status. This uses a group of alert comments to determine which sub cases should
+ * be retrieved. This is based on whether the comment is associated to a sub case.
+ */
+async function getSubCasesToStatus({
+  totalAlerts,
+  caseService,
+  client,
+}: {
+  totalAlerts: SavedObjectsFindResponse<CommentAttributes>;
+  caseService: CaseServiceSetup;
+  client: SavedObjectsClientContract;
+}): Promise<Map<string, CaseStatuses>> {
+  const subCasesToRetrieve = totalAlerts.saved_objects.reduce((acc, alertComment) => {
+    if (
+      isCommentRequestTypeAlertOrGenAlert(alertComment.attributes) &&
+      alertComment.attributes.associationType === AssociationType.subCase
+    ) {
+      const id = getID(alertComment, SUB_CASE_SAVED_OBJECT);
+      if (id !== undefined) {
+        acc.add(id);
+      }
+    }
+    return acc;
+  }, new Set<string>());
+
+  const subCases = await caseService.getSubCases({
+    ids: Array.from(subCasesToRetrieve.values()),
+    client,
+  });
+
+  return subCases.saved_objects.reduce((acc, subCase) => {
+    // log about the sub cases that we couldn't find
+    if (!subCase.error) {
+      acc.set(subCase.id, subCase.attributes.status);
+    }
+    return acc;
+  }, new Map<string, CaseStatuses>());
+}
+
+/**
+ * Returns what status the alert comment should have based on whether it is associated to a case or sub case.
+ */
+function getSyncStatusForComment({
+  alertComment,
+  casesToSyncToStatus,
+  subCasesToStatus,
+}: {
+  alertComment: SavedObjectsFindResult<CommentAttributes>;
+  casesToSyncToStatus: Map<string, CaseStatuses>;
+  subCasesToStatus: Map<string, CaseStatuses>;
+}): CaseStatuses {
+  let status: CaseStatuses = CaseStatuses.open;
+  if (alertComment.attributes.associationType === AssociationType.case) {
+    const id = getID(alertComment, CASE_SAVED_OBJECT);
+    // We should log if we can't find the status
```

Comment:
Another log comment. I agree, maybe a parameter sent with the response `errors: ['We were unable to retrieve x, y, and z']`

Reply:
Yeah I'll add the logging in a follow up PR. I need to pass the plugin's logger around to do it I think.